File: backend/app/schemas/team.py
"""
团队管理相关的Pydantic模型
"""

from typing import Optional, List
from datetime import datetime
from pydantic import BaseModel, EmailStr


class TeamCreate(BaseModel):
    """创建团队的请求模型"""
    name: str
    description: Optional[str] = None
    team_type: Optional[str] = "collaborative"  # personal, collaborative, project
    max_members: Optional[int] = 100
    is_private: Optional[bool] = True


class TeamUpdate(BaseModel):
    """更新团队的请求模型"""
    name: Optional[str] = None
    description: Optional[str] = None
    team_type: Optional[str] = None
    max_members: Optional[int] = None
    is_private: Optional[bool] = None


class TeamResponse(BaseModel):
    """团队信息响应模型"""
    id: int
    name: str
    description: Optional[str] = None
    team_type: str
    max_members: int
    member_count: int
    created_by: Optional[int] = None
    created_at: datetime
    my_role: Optional[str] = None
    my_member_type: Optional[str] = None
    is_private: bool

    class Config:
        from_attributes = True


class TeamMemberResponse(BaseModel):
    """团队成员信息响应模型"""
    user_id: int
    username: str
    email: str
    full_name: Optional[str] = None
    role: str
    member_type: str
    join_time: datetime
    invited_by: Optional[int] = None

    class Config:
        from_attributes = True


class TeamInvitationCreate(BaseModel):
    """创建团队邀请的请求模型"""
    email: EmailStr
    target_role: Optional[str] = "USER"
    target_member_type: Optional[str] = "member"
    message: Optional[str] = None


class TeamInvitationResponse(BaseModel):
    """团队邀请响应模型"""
    id: str
    team_id: int
    team_name: str
    inviter_id: int
    inviter_name: str
    invitee_email: str
    invite_code: str
    target_role: str
    target_member_type: str
    message: Optional[str] = None
    expire_time: Optional[datetime] = None
    status: str
    create_time: datetime

    class Config:
        from_attributes = True


class TeamMemberUpdate(BaseModel):
    """更新团队成员的请求模型"""
    role: Optional[str] = None
    member_type: Optional[str] = None


class JoinTeamRequest(BaseModel):
    """加入团队的请求模型"""
    invite_code: str

class TeamJoinRequest(BaseModel):
    """加入团队的请求模型（别名）"""
    invite_code: str


class TeamSwitchRequest(BaseModel):
    """切换团队的请求模型"""
    team_id: int


class TeamPermissionCheck(BaseModel):
    """团队权限检查的请求模型"""
    user_id: int
    team_id: int
    permission: str


class TeamStats(BaseModel):
    """团队统计信息"""
    total_members: int
    active_members: int
    knowledge_bases: int
    documents: int
    storage_used_mb: int
    storage_quota_mb: int

    class Config:
        from_attributes = True

File: frontend/src/types/team.ts
/**
 * 团队管理相关的TypeScript类型定义
 */

export interface Team {
  id: number;
  name: string;
  description?: string;
  team_type: 'personal' | 'collaborative' | 'project';
  max_members: number;
  member_count: number;
  created_by?: number;
  created_at: string;
  my_role?: string;
  my_member_type?: 'owner' | 'admin' | 'member';
  is_private: boolean;
}

export interface TeamMember {
  user_id: number;
  username: string;
  email: string;
  full_name?: string;
  role: string;
  member_type: 'owner' | 'admin' | 'member';
  join_time: string;
  invited_by?: number;
}

export interface TeamInvitation {
  id: string;
  team_id: number;
  team_name: string;
  inviter_id: number;
  inviter_name: string;
  invitee_email: string;
  invite_code: string;
  target_role: string;
  target_member_type: string;
  message?: string;
  expire_time?: string;
  status: '0' | '1' | '2'; // 0-过期 1-有效 2-已使用
  create_time: string;
}

export interface CreateTeamData {
  name: string;
  description?: string;
  team_type?: 'personal' | 'collaborative' | 'project';
  max_members?: number;
  is_private?: boolean;
}

export interface UpdateTeamData {
  name?: string;
  description?: string;
  team_type?: 'personal' | 'collaborative' | 'project';
  max_members?: number;
  is_private?: boolean;
}

export interface InviteUserData {
  email: string;
  target_role?: string;
  target_member_type?: 'owner' | 'admin' | 'member';
  message?: string;
}

export interface TeamStats {
  total_members: number;
  active_members: number;
  knowledge_bases: number;
  documents: number;
  storage_used_mb: number;
  storage_quota_mb: number;
}

export interface TeamPermission {
  can_read: boolean;
  can_write: boolean;
  can_create: boolean;
  can_delete: boolean;
  can_manage: boolean;
  can_invite: boolean;
  can_remove_members: boolean;
}

export const TEAM_TYPES = {
  PERSONAL: 'personal',
  COLLABORATIVE: 'collaborative',
  PROJECT: 'project',
} as const;

export const MEMBER_TYPES = {
  OWNER: 'owner',
  ADMIN: 'admin',
  MEMBER: 'member',
} as const;

export const TEAM_ROLES = {
  OWNER: 'OWNER',
  ADMIN: 'ADMIN',
  USER: 'USER',
} as const;